flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchSinkBuilderBase.java

```diff
@@ -23,31 +24,21 @@
 
 import org.apache.http.HttpHost;
 
+import java.io.Serializable;
 import java.util.Arrays;
 import java.util.List;
 
+import static org.apache.flink.util.Preconditions.checkArgument;
 import static org.apache.flink.util.Preconditions.checkNotNull;
 import static org.apache.flink.util.Preconditions.checkState;
 
 /**
- * Builder to construct a {@link ElasticsearchSink}.
- *
- * <p>The following example shows the minimal setup to create a ElasticsearchSink that submits
- * actions on checkpoint or the default number of actions was buffered (1000).
- *
- * <pre>{@code
- * Elasticsearch<String> sink = Elasticsearch
- *     .builder()
- *     .setHosts(MY_ELASTICSEARCH_HOSTS)
- *     .setEmitter(MY_ELASTICSEARCH_EMITTER)
- *     .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
- *     .build();
- * }</pre>
+ * Base builder to construct a {@link ElasticsearchSink}.
  *
  * @param <IN> type of the records converted to Elasticsearch actions
  */
 @PublicEvolving
-public class ElasticsearchSinkBuilder<IN> {
+public abstract class ElasticsearchSinkBuilderBase<IN> implements Serializable {
```

Comment:
For base builders, you usually need to capture the type as well `ElasticsearchSinkBuilderBase<IN, B extends ElasticsearchSinkBuilderBase<IN>>` in case you want to add ES6 / 7 specific methods later.

Reply:
Why is the builder serializable?

Reply:
Regarding the base builder I initially tried the type capture but could not get `setEmitter` to work with in this setup. If you could help me out there we can try to change it

The serializable was an accident, I'll remove it.